### PR TITLE
Fixed UnitTesting issues with NitroxTest

### PR DESCRIPTION
**Pull request by killzoms**

### CODE REVIEW

File: NitroxPatcher/Patches/BaseGhost_Finish_Patch.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public static IEnumerable<CodeInstruction> Transpiler(MethodBase original, IEnum
                      * TransientLocalObjectManager.Add(TransientLocalObjectManager.TransientObjectType.BASE_GHOST_NEWLY_CONSTRUCTED_BASE_GAMEOBJECT, gameObject);
                      */
                     yield return new CodeInstruction(OpCodes.Ldc_I4_1);
-                    yield return original.Ldloc<GameObject>(1);
+                    yield return original?.Ldloc<GameObject>(1);
```
Comment (Sunrunner37) [inline NitroxPatcher/Patches/BaseGhost_Finish_Patch.cs]: These should not be nullable.  We need an exception or they will silently fail if UWE changes them.

File: NitroxClient/Communication/Packets/Processors/BedEnterProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxModel.Packets;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    class BedEnterProcessor : ClientPacketProcessor<BedEnter>
```
Comment (Sunrunner37) [inline NitroxClient/Communication/Packets/Processors/BedEnterProcessor.cs]: Let's remove and add an issue instead.

File: NitroxTest/Model/PacketProcessorTest.cs [hunk 3]
```diff
@@ -55,8 +55,20 @@ public void RuntimeDetectsAllClientPacketProcessors()
...
-                        $"{processor} has not been discovered by the runtime code!")
+                {
+                    try
+                    {
+                        Assert.IsTrue(clientDependencyContainer.ResolveOptional(processor.BaseType) != null,
+                            $"{processor} has not been discovered by the runtime code!");
+                    }
+                    catch (Autofac.Core.DependencyResolutionException ex)
+                    {
+                        if (ex.InnerException.GetType() != typeof(System.Security.SecurityException))
```
Comment (Sunrunner37) [inline NitroxTest/Model/PacketProcessorTest.cs]: What components are throwing e call violations?

File: NitroxPatcher/Patches/BaseGhost_Finish_Patch.cs (hunk 1 above)
Comment (killzoms) [reply NitroxPatcher/Patches/BaseGhost_Finish_Patch.cs]: Made it that way due too the unitTest passing in a null parameter...

File: NitroxClient/Communication/MultiplayerSession/MultiplayerSessionManager.cs [hunk 4]
```diff
@@ -78,7 +78,6 @@ public void RequestSessionReservation(PlayerSettings playerSettings, Authenticat
         public void ProcessReservationResponsePacket(MultiplayerSessionReservation reservation)
         {
             Reservation = reservation;
-            CurrentState.NegotiateReservation(this);
```
Comment (Sunrunner37) [inline NitroxClient/Communication/MultiplayerSession/MultiplayerSessionManager.cs]: Should we be removing this?

Comment (MadMaxOfYore) [reply NitroxClient/Communication/MultiplayerSession/MultiplayerSessionManager.cs]: Sorry @killzoms but SR is correct. This is the state that that session manager should be in upon receiving a confirmed reservation:

```suggestion
            CurrentState.NegotiateReservation(this);
```
https://github.com/SubnauticaNitrox/Nitrox/blob/master/NitroxClient/Communication/MultiplayerSession/ConnectionState/AwaitingSessionReservation.cs#L8

**Comment by killzoms** [comment]
figured out how to update local submodules with the latest commit...

**Comment by killzoms** [comment]
Considering the age I will be closing this merge request

**Pull request closed without merging**